### DISCUSSION TITLE
support proxy redirect

### DISCUSSION THREAD

**Discussion by mixxen** [chimurai/http-proxy-middleware#546] [Ideas]
Your rewrite library seems to be working well. It would be nice if your library supported rewriting proxy redirects. For instance, a proxy is configured like so:

```JavaScript
  autoRewrite: true,
  pathRewrite: {
    '^/newpath/' : '/'
  }
```

If the target server responds with 302, then the client will be redirected to a 404 page because ```/newpath/``` is not appended to the location header. To workaround this, I have modified ```web-outgoing.js``` in node_modules/http-proxy to check if ```autoRewrite``` is a string. If it is a string, append it to the location. I'm sure there is a better way to do this....

```JavaScript
  autoRewrite: '/newpath',
  pathRewrite: {
    '^/newpath/' : '/'
  }
```
```JavaScript
  if (options.hostRewrite) {
    u.host = options.hostRewrite;
  } else if (options.autoRewrite) {
    if(typeof options.autoRewrite === 'string')
      u.host = req.headers['host'] + options.autoRewrite;
    else
      u.host = req.headers['host'];
  }

```

**Comment by chimurai** [comment]
Thanks for reporting the issue @mixxen 

I'm afraid I won't have much time to investigate and create a fix the coming weeks.
Care to take a shot at creating a PR?

**Comment by roccomuso** [comment]
+1
